scripts/updateMarketplaceReadme.mts

```diff
@@ -0,0 +1,79 @@
+import * as path from "node:path";
+import * as fs from "fs";
+import { GraphQLClient, gql } from "graphql-request";
+
+const pdClient = new GraphQLClient(
+  "https://api.pipedream.com/graphql",
+  {
+    headers: {
+      Authorization: `Bearer ${process.env.PD_API_KEY}`,
+    },
+  }
+);
+
+  // This script expects the first argument to be a comma separated list of file paths.
+  // Example: `ts-node scripts/updateMarketplaceReadme.mts components/slack/README.md,components/slack/actions/add-star/README.md`
+const run = async () => {
+  const fileCsv = process.argv[2] || "";
+
+  const filePaths = fileCsv.split(",").filter(Boolean);
+
+  const readmePaths = filePaths.filter(
+    (p) => p.startsWith("components/") && p.endsWith("/README.md")
+  );
+
+  for (const readmePath of readmePaths) {
+    const fullPath = path.join(process.cwd(), readmePath);
+
+    console.log("processing file", fullPath);
+
+    const b64 = fs.readFileSync(fullPath).toString("base64");
+
+    const pathSegments = readmePath.split("/");
+
+    let key: string;
+
+    if (pathSegments.length === 3) {
+      key = pathSegments[1];
+    } else if (pathSegments.length === 5) {
+      key = pathSegments[1] + "-" + pathSegments[3];
+    } else {
+      // TODO handle invalid file path
+      console.warn(
+        `"${readmePath}" is an invalid path. Cannot determine if this is an app or a component. Skipping...`
+      );
```

Comment:
Currently, I'm not doing anything for invalid readme paths. Should we be pushing an alert anywhere for this, or just log it and move on?

Reply:
These warns will likely not be seen -- it might be easy to create a sentry app for this public repo and then provide the key via secrets and just `Sentry.captureMessage`?

Reply:
(if you do this i recommend making the message static and passing the variable as an `extra: {...}` so that the grouping is easier maybe)